src/main/java/com/aws/greengrass/certificatemanager/certificate/CISShadowMonitor.java

```diff
@@ -243,20 +247,37 @@ private synchronized void handleNewCloudVersion(int newVersion) {
                 for (CertificateGenerator cg : monitoredCertificateGenerators) {
                     cg.generateCertificate(connectivityInfoProvider::getCachedHostAddresses);
                 }
-                reportVersion(version);
-                lastVersion = version;
+
+                try {
+                    reportVersion(version);
+                } catch (ExecutionException e) {
+                    Throwable cause = e.getCause() == null ? e : e.getCause();
+                    LOGGER.atWarn().kv("version", version).cause(cause).log("Unable to report CIS shadow version");
+                } catch (TimeoutException e) {
+                    LOGGER.atWarn().kv("version", version).log("Timed out while reporting CIS shadow version");
+                } catch (InterruptedException e) {
+                    LOGGER.atWarn().kv("version", version).log("Interrupted while reporting CIS shadow version");
+                } catch (Exception e) {
+                    LOGGER.atWarn().kv("version", version).cause(e).log("Unable to report CIS shadow version");
+                } finally {
+                    lastVersion = version;
+                }
             } catch (KeyStoreException e) {
                 LOGGER.atError().cause(e).log("failed to generate new certificates");
             }
         });
     }
 
-    private void reportVersion(int version) {
+    private void reportVersion(int version) throws ExecutionException, InterruptedException, TimeoutException {
         LOGGER.atInfo().kv("version", version).log("Reporting version");
         UpdateShadowRequest updateShadowRequest = new UpdateShadowRequest();
         updateShadowRequest.thingName = shadowName;
         updateShadowRequest.version = version;
-        iotShadowClient.PublishUpdateShadow(updateShadowRequest, QualityOfService.AT_LEAST_ONCE);
+        updateShadowRequest.state = new ShadowState();
+        updateShadowRequest.state.reported = new HashMap<>();
+        updateShadowRequest.state.desired = new HashMap<>();
+        iotShadowClient.PublishUpdateShadow(updateShadowRequest, QualityOfService.AT_LEAST_ONCE)
+                .get(5, TimeUnit.SECONDS);
```

Comment:
I don't think we should be waiting here, this is called in a thread that shouldn't be blocked. You can chain the future to run code on success or error

Reply:
@MikeDombo that's not true. We're calling this from an executor service, not the CRT thread